edb/pgsql/trampoline.py

```diff
@@ -0,0 +1,134 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2016-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+"""Support for namespacing and trampolining the standard library.
+
+The idea here is that all of the functions, tables, and views in
+edgedb/edgedbstd/edgedbsql should be moved into namespaced libraries
+of the form `edgedb_VER`, where VER will be substituted with some
+version identifier.
+
+Then, for anything that might be referenced by a function, constraint,
+etc, we will create a *trampoline* in the un-suffixed namespace.
+When doing (eventually) in-place version upgrades, we will create the
+new namespace and then update the trampolines to point to it.
+
+CURRENT STATUS: So far, functions and views are mostly
+namespaced. Standard library schema object tables aren't yet.
+"""
+
+from __future__ import annotations
+from typing import (
+    Optional,
+    TYPE_CHECKING,
+)
+
+import copy
+
+
+from edb import buildmeta
+
+from . import common
+from . import dbops
+
+
+q = common.qname
+qi = common.quote_ident
+
+
+def versioned_schema(s: str, version: Optional[int]=None) -> str:
+    if version is None:
+        # ... get_version_dict() is cached
+        version = buildmeta.get_version_dict()['major']
+    return f'{s}_v{version}'
```

Comment:
Since we're not consistently quoting schema names in SQL, I'd add a comment here that the return value MUST be lowercase.